server/utils/next_url.ts

```diff
@@ -25,8 +25,20 @@ export function composeNextUrlQeuryParam(request: KibanaRequest, basePath: strin
   return stringify({ nextUrl });
 }
 
+export const INVALID_NEXT_URL_PARAMETER_MESSAGE = 'Invalid nextUrl parameter.';
+
+/**
+ * Ensures the nextUrl parameter is a relative url, the nextUrl parameter should:
+ *   1. starts with '/'
+ *   2. not start with '//'
+ *   3. does not contain '@' in the path
+ * @param url
+ */
 export const validateNextUrl = (url: string | undefined): string | void => {
-  if (url && url.toLowerCase().includes('//')) {
-    return 'invalid nextUrl parameter';
+  if (url) {
+    const path = url.split('?')[0];
+    if (!path.startsWith('/') || path.startsWith('//') || path.includes('@')) {
```

Comment:
There is no url decode process here, if user use `%2F` to represent `/`, will it bypass this validation?

Reply:
E.g. `/%2Fexample.com`

Reply:
url decode is automatically done by Kibana web server, so no need to decode it here again